File: ios/SpentyAI/SpentyAI/Core/Models/Vendor.swift
import Foundation

struct Vendor: Codable, Identifiable {
    let id: String
    var name: String?
    var email: String?
    var phone: String?
    var gstin: String?
    var billingAddress: String?
    var totalBilled: Double?
    var totalPaid: Double?
    var outstanding: Double?

    enum CodingKeys: String, CodingKey {
        case id = "vendorId"
        case name
        case email
        case phone
        case gstin
        case billingAddress
        case totalBilled
        case totalPaid
        case outstanding
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/models/Vendor.kt
package com.spentyai.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Vendor(
    @SerialName("vendor_id") val id: String,
    val name: String,
    val email: String? = null,
    val phone: String? = null,
    val company: String? = null,
    val address: Address? = null,
    @SerialName("tax_id") val taxId: String? = null,
    val notes: String? = null,
    @SerialName("is_active") val isActive: Boolean = true,
    @SerialName("total_billed") val totalBilled: Double = 0.0,
    @SerialName("total_paid") val totalPaid: Double = 0.0,
    @SerialName("outstanding_balance") val outstandingBalance: Double = 0.0,
    val currency: String = "USD",
    @SerialName("payment_terms") val paymentTerms: String? = null,
    @SerialName("default_category_id") val defaultCategoryId: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)
